lib/solr.ts

```diff
@@ -738,7 +738,7 @@ get(handler: string, query: any, callback?: any) {
    * @return {http.ClientRequest}
    * @api public
    */
-post(handler, query?: Query | Record<string, any> | string, callback?: CallbackFn) {
+post(handler, callback: CallbackFn, query?: Query | Record<string, any> | string, ) {
```

Comment:
This is a breaking change that will require all users of the library to change their code. It is generally better to avoid such changes without a very good reason. I don't think this change was necessary - instead there seems to be one missing type for `query` param, judging by the code, it can actually be a CallbackFn itself.